app/main.py

```diff
@@ -1 +1,31 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = hidden
+        if self.health > 0:
+            Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(herbivore: Herbivore) -> None:
+        if herbivore.hidden is False and isinstance(herbivore, Herbivore):
+            herbivore.health -= 50
+            if herbivore.health < 1:
+                herbivore.alive.remove(herbivore)
```

Comment:
Read the checklist:

![image](https://user-images.githubusercontent.com/104218550/218091686-a081f73e-2194-4472-acc5-a6768561944a.png)

Reply:
Done